nodejs/week1/src/backend/routes/cheap-meals.js

```diff
@@ -0,0 +1,19 @@
+const express = require("express");
+const dataAccess = require("../dataAccess.js");
+const router = express.Router();
+const meals = "/data/meals.json";
+const reviews = "/data/reviews.json";
+
+router.get("/cheap-meals", (request, response) => {
+  let mealsData = dataAccess.getDataFromJson(meals);
+  let reviewsData = dataAccess.getDataFromJson(reviews);
+  let mealsWithReviews = mealsData.map((meal) => {
+    meal.review = reviewsData.filter((review) => review.id === meal.id);
+    return meal;
+  });
+  let cheapMealsWithReviews = mealsWithReviews.filter(
+    (meal) => meal.price < 70
+  );
+  response.send(cheapMealsWithReviews);
```

Comment:
Nice! This is a great use of map and filter. I also like how you are using very descriptive names for things. Yey!

Hint: This could be extracted into a function that gives you mealsAndReviews. Consider putting this function in its own file and using it in all other routers that need meals with reviews.

In general, try to break down problems into their small parts:

e.g. Get cheap meals with reviews
Step 1) Get meals with reviews
Step 2) Get cheap meals with reviews

When you break down problems like this, you'll find that step 1) is repeated across many other problems:

Get larger meals (including reviews): 
Step 1) Get meals with reviews
Step 2) Get larger meals with reviews

Get random meal (including reviews):
Step 1) Get meals with reviews
Step 2) Get one random meal with reviews

When you find yourself writing the same code more than twice, consider whether you can extract a helper function that you can reuse from both places. This principle is commonly known as DRY (Don't Repeat Yourself). Keeping your code DRY helps you right more maintainable code because:

If there's a bug in this code, it's only in one place and you only need to fix it once
When you create a helper function other team mates can benefit from it and reuse your code
When someone looks at your code they need to read it slowly and understand what is going on. When they read a helper function like getRandomInteger(maxValue) they automatically understand what the function does without needing to exert their cognitive abilities a lot.